src/uu/wc/src/count_fast.rs

```diff
@@ -87,7 +92,12 @@ pub(crate) fn count_bytes_fast<T: WordCountable>(handle: &mut T) -> (usize, Opti
             // example with /proc/version and /sys/kernel/profiling. So,
             // if it is 0 we don't report that and instead do a full read.
             if (stat.st_mode as libc::mode_t & S_IFREG) != 0 && stat.st_size > 0 {
-                return (stat.st_size as usize, None);
+                if let Ok(offset) = is_stdin
+                    .then(|| lseek(fd, 0, nix::unistd::Whence::SeekCur))
+                    .unwrap_or(Ok(0))
+                {
+                    return (stat.st_size.sub(offset).max(0) as usize, None);
+                }
```

Comment:
Would this make sense, i.e. get rid of the `if let`, so that we just return the size if the `lseek` fails?
```rust
let offset = if is_stdin {
    lseek(fd, 0, nix::unistd::Whence::SeekCur).unwrap_or(0)
} else {
    0
};
return (stat.st_size.sub(offset).max(0) as usize, None);
```

Also could you explain in a comment why this seek is necessary?